src/ucp/proto/proto_am.inl

```diff
@@ -174,6 +174,8 @@ ucs_status_t ucp_do_am_zcopy_single(uct_pending_req_t *self, uint8_t am_id,
         req->send.state = saved_state; /* need to restore the offsets state */
         return status;
     } else {
+        ++req->send.uct_comp.count;
```

Comment:
this is not the same, dt_state.offset does not influence on whether a completion callback will be called, but uct_comp.cont does. this is why ++comp_count is done before (so callback will not be called in the theoretical case of multi-thread). but it's ok to do dt_state.offset+=length afterwards